### PR TITLE
[Storage] Add warning when pinning a cid twice

### LINKED ISSUE #1212: ChainSafe Storage: User can pin the same file multiple times
**Description**
It is possible for the user to repeatedly pin the same file. 

**To Reproduce**
1. Login to ChainSafe storage 
2. Navigate to the Cids page
3. Click on the Pin CID button
4. Paste a CID into the input box
5. Click Pin
6. Repeat steps 2 -5 with the same CID

**Expected behavior**
I don't think there's a use case for this behaviour so perhaps it would be better to inform the user that they have already pinned that CID?

**Screenshots**
Screen Shot 2021-06-30 at 11 57 15 AM

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser Chrome version 91.0.4472.114, Firefox version 89.0.2

### PR DESCRIPTION
**Pull request by Tbaut**
closes #1212

### CODE REVIEW

**Review by asnaith — APPROVED** [review]
Good solution for the scenario, warning over prevention makes sense :)

File: packages/storage-ui/src/Components/Modules/AddCIDModal.tsx [hunk 1]
```diff
@@ -63,97 +75,136 @@ interface IAddCIDModuleProps {
...
+          setShowWarning(true)
+        }
+      })
+  }, [cid, searchCid])
+
+  const onCidChange = useCallback((cid?: string | number) => {
+    setCidError("")
+    setShowWarning(false)
+
+    cid && setCid(cid?.toString())
```
Comment (tanmoyAtb) [inline packages/storage-ui/src/Components/Modules/AddCIDModal.tsx]: This is preventing erasing the first character of the input. This should fix it.
```suggestion
    setCid(cid?.toString() || "")
```

**Review by tanmoyAtb — CHANGES_REQUESTED** [review]
So i tried this approach ->
I copied a CID without the last character,
typed in the last character
but I did not get the warning.

While we are at it, On a UI note, I would prefer the CID modal to be wider than what we have.

File: packages/storage-ui/src/Components/Modules/AddCIDModal.tsx [hunk 2]
```diff
@@ -63,97 +75,136 @@ interface IAddCIDModuleProps {
...
+    cidValidator
+      .validate({ cid })
+      .then(() => {
+        setCidError("")
+        warnforDuplicatesCid()
+      })
+      .catch((e: Error) => {
+        setCidError(e.message)
+      })
+  }, [warnforDuplicatesCid])
```
Comment (FSM1) [inline packages/storage-ui/src/Components/Modules/AddCIDModal.tsx]:
```suggestion
  const checkForDuplicateCid = useCallback(() => {
    searchCid(cid)
      .then((res) => {
        if(res && res.results?.length){
          setShowWarning(true)
        }
      })
  }, [cid, searchCid])

  const onCidChange = useCallback((cid?: string | number) => {
    setCidError("")
    setShowWarning(false)

    setCid(cid?.toString() || "")

    cidValidator
      .validate({ cid })
      .then(() => {
        setCidError("")
        checkForDuplicateCid()
      })
      .catch((e: Error) => {
        setCidError(e.message)
      })
  }, [checkForDuplicateCid])
```

File: packages/storage-ui/src/Components/Modules/AddCIDModal.tsx [hunk 3]
```diff
@@ -63,97 +75,136 @@ interface IAddCIDModuleProps {
...
-                loading={accessingCID}
-                data-cy="button-submit-pin"
-              >
-                <Trans>Pin</Trans>
-              </Button>
-            </Grid>
-          </div>
-        </Form>
-      </Formik>
+              <Trans>Warning: cid already pinned</Trans>
```
Comment (FSM1) [inline packages/storage-ui/src/Components/Modules/AddCIDModal.tsx]:
```suggestion
              <Trans>Warning: CID already pinned</Trans>
```

**Review by FSM1 — APPROVED** [review]
Just a few cosmetic changes, but working great otherwise 👍🏼

File: packages/storage-ui/src/Components/Modules/AddCIDModal.tsx (hunk 3 above)
Comment (Tbaut) [reply packages/storage-ui/src/Components/Modules/AddCIDModal.tsx]: :+1:

File: packages/storage-ui/src/Components/Modules/AddCIDModal.tsx (hunk 2 above)
Comment (Tbaut) [reply packages/storage-ui/src/Components/Modules/AddCIDModal.tsx]: refactored that part :)

**Comment by Tbaut** [comment]
Thanks Tanmoy and Mike. My logic was a little meh for the check. I re-implemented that in a nicer way. The problem you mentioned @tanmoyAtb is now fixed, great catch.

**Comment by Tbaut** [comment]
BTW I agree with the modal style that's a little weird. I'll change that when adding a name together with https://github.com/ChainSafe/ui-monorepo/issues/2022

**Review by tanmoyAtb — APPROVED** [review]
Looking 💯

**Pull request merged**